pkg/types/azure/disk.go

```diff
@@ -0,0 +1,65 @@
+package azure
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/pkg/errors"
+)
+
+var (
+	RxDiskEncryptionSetID   = regexp.MustCompile(`(?i)^/subscriptions/([0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})/resourceGroups/([-a-z0-9_().]{0,89}[-a-z0-9_()])/providers/Microsoft\.Compute/diskEncryptionSets/([-a-z0-9_]{1,80})$`)
+	RxSubscriptionID        = regexp.MustCompile(`^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$`)
+	RxResourceGroup         = regexp.MustCompile(`^[-a-z0-9_().]{0,89}[-a-z0-9_()]$`)
+	RxDiskEncryptionSetName = regexp.MustCompile(`^[-a-z0-9_]{1,80}$`)
+)
+
+// DiskEncryptionSet defines the configuration for a disk encryption set.
+type DiskEncryptionSet struct {
+	SubscriptionID string `json:"subscriptionId"`
+	ResourceGroup  string `json:"resourceGroup"`
+	Name           string `json:"name"`
+}
+
+// ToID creates an Azure resource ID for the disk encryption set.
+func (d *DiskEncryptionSet) ToID() string {
+	return fmt.Sprintf("/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Compute/diskEncryptionSets/%s",
+		d.SubscriptionID, d.ResourceGroup, d.Name)
+}
+
+// Set fills a DiskEncryptionSet struct from a disk encryption set ID string
+func (d *DiskEncryptionSet) Set(ID string) error {
```

Comment:
I'm not following this: why are we setting from the ID rather than `*DiskEncryptionSet`?  If this is only used in tests it should be moved to tests, but I think it makes more sense to have this Set from `*DiskEncryptionSet` and used by https://github.com/openshift/installer/blob/master/pkg/types/azure/machinepool.go#L43 with a pattern similar to the GCP encryption keys: https://github.com/openshift/installer/blob/master/pkg/types/gcp/machinepools.go#L65-L70

Reply:
Nuked.